src/main/java/com/palantir/gradle/versions/GradleConfigurations.java

```diff
@@ -16,14 +16,23 @@
 
 package com.palantir.gradle.versions;
 
+import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableSet;
 import java.util.Set;
 import java.util.stream.Stream;
 import org.gradle.api.Project;
 import org.gradle.api.artifacts.Configuration;
 import org.gradle.api.plugins.JavaPluginConvention;
+import org.gradle.api.tasks.SourceSet;
 
 final class GradleConfigurations {
+    /**
+     * Deprecated sourcesets of the compile and runtime configuration, to be removed with Gradle 7.
+     * The full configuration name follows the naming scheme of "$taskBaseName + capitalize($configurationName)"
+     * see {@link org.gradle.api.internal.tasks.DefaultSourceSet#configurationNameOf}.
+     */
+    private static final ImmutableList<String> DEPRECATED_SOURCESET_SUFFIXES = ImmutableList.of("Compile", "Runtime");
```

Comment:
The `getCompileConfigurationName` and `getRuntimeConfigurationName` method no longer exist with Gradle 7 and the method for building the configuration name based on the base config is private ([ref](https://github.com/gradle/gradle/blob/a2ea10896cbdca85a4fb7b0e30d638f57dbe1533/subprojects/plugins/src/main/java/org/gradle/api/internal/tasks/DefaultSourceSet.java#L143-L145)).

Matching on the configuration name suffix for backwards compatibility but for Gradle 7 projects, this shouldn't match anything and can be removed in the future.